academic_observatory_workflows/workflows/tests/test_data_quality_workflow.py

```diff
@@ -0,0 +1,544 @@
+
+        super(TestDataQualityWorkflow, self).__init__(*args, **kwargs)
+
+    def test_dag_load(self):
+        """Test that the DataQualityCheck DAG can be loaded from a DAG bag."""
+
+        env = ObservatoryEnvironment(
+            workflows=[
+                Workflow(
+                    dag_id=self.dag_id,
+                    name="Data Quality Check Workflow",
+                    class_name="academic_observatory_workflows.workflows.data_quality_workflow.DataQualityWorkflow",
+                    cloud_workspace=self.fake_cloud_workspace,
+                    kwargs=dict(
+                        sensor_dag_ids=["doi"],
+                        datasets={
+                            "observatory": {"tables": [{"table_id": "doi", "is_sharded": True, "fields": ["doi"]}]}
+                        },
+                    ),
+                )
+            ]
+        )
+
+        with env.create():
+            self.assert_dag_load_from_config(self.dag_id)
+
+    def test_dag_structure(self):
+        """Test that the DAG has the correct structure."""
+
+        workflow = DataQualityWorkflow(
+            dag_id=self.dag_id,
+            cloud_workspace=self.fake_cloud_workspace,
+            sensor_dag_ids=["dummy1", "dummy2"],
+            datasets={
+                "observatory": {"tables": [{"table_id": "doi", "is_sharded": True, "fields": ["doi"]}]},
+                "pubmed": {
+                    "tables": [
+                        {
+                            "table_id": "pubmed",
+                            "is_sharded": False,
+                            "fields": ["MedlineCitation.PMID.value", "MedlineCitation.PMID.Version"],
+                        }
+                    ]
+                },
+            },
+        )
+        dag = workflow.make_dag()
+        self.assert_dag_structure(
+            {
+                "dag_sensors.dummy1_sensor": ["check_dependencies"],
+                "dag_sensors.dummy2_sensor": ["check_dependencies"],
+                "check_dependencies": ["create_dataset"],
+                "create_dataset": ["observatory.doi", "pubmed.pubmed"],
+                "observatory.doi": [],
+                "pubmed.pubmed": [],
+            },
+            dag,
+        )
+
+    def test_workflow(self):
+        """Test the Data Quality Check Workflow end to end
+
+        Borrowing off of the doi test structure."""
+
+        env = ObservatoryEnvironment(self.project_id, self.data_location)
+
+        # Where the metadata generated for this workflow is going to be stored.
+        dqc_dataset_id = env.add_dataset(prefix="data_quality_check")
+        fake_dataset_id = env.add_dataset()
+        settings_dataset_id = env.add_dataset(prefix="settings")
+
+        # Borrowing from the DOI Workflow to create fake data for these tests
+        repositories = TestDoiWorkflow().repositories
+        institutions = TestDoiWorkflow().institutions
```

Comment:
It is better not to import TestDoiWorkflow like this, because at least with PyCharm, it will then also run the TestDoiWorkflow tests as well.

It would be better to load a dataset that you can use to check that the data quality workflow is producing correct results, you could even just write it by hand in JSON files, since you really just need an ID field for each table. You want to test that the workflow can operate on sharded and non sharded tables, single fields, and mutliple nested fields (like the pubmed example). Then also test that the output is correct - so have some duplicate IDs, null ids etc, which might not be present in the fake DOI Workflow data.